google/resource_compute_instance.go

```diff
@@ -378,22 +378,39 @@ func resourceComputeInstance() *schema.Resource {
 
 			"scheduling": &schema.Schema{
 				Type:     schema.TypeList,
+				MaxItems: 1,
 				Optional: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					// The `scheduling` block is optional in the config but implicitly has a length
+					// of 1 with the default values for `automatic_restart` and `preemptible`.
+					// This suppress the diff `scheduling.#: "1" => "0"` when the `scheduling` block
+					// is not specified by the user.
+					if k == "scheduling.#" {
```

Comment:
Not exactly,

If the user don't add the `scheduling` block to the config because they are fine with the default values. Then, the desired state as scheduling.# = 0.

But the current state gets refreshed (calls the read method), the populates the values for the current state scheduling. Hence, `scheduling.#` will always be 0 for the current state.

This means a user gets a perpetual diff if they don't specify the scheduling block (which is an okay thing to do if you are fine with the default values):
 `scheduling.#: "1" => "0"`